internal/transport/http_util.go

```diff
@@ -267,49 +274,19 @@ func (d *decodeState) decodeHeader(frame *http2.MetaHeadersFrame) error {
 		return status.Error(codes.Internal, "peer header list size exceeded limit")
 	}
 
-	// isGRPC indicates whether the peer is speaking gRPC (otherwise HTTP).
-	//
-	// We are in gRPC mode (peer speaking gRPC) if:
-	// 	* We are client side and have already received a HEADER frame that indicates gRPC peer.
-	//  * The header contains valid  a content-type, i.e. a string starts with "application/grpc"
-	// And we should handle error specific to gRPC.
-	//
-	// Otherwise (i.e. a content-type string starts without "application/grpc", or does not exist), we
-	// are in HTTP fallback mode, and should handle error specific to HTTP.
-	//
-	// d.ignoreContentType is only set on client side after a gRPC ResponseHeader has been received (indicating
-	// peer speaking gRPC). Therefore, we can initialized isGRPC to d.ignoreContentType.
-	isGRPC := d.ignoreContentType
-	var grpcErr, httpErr, contentTypeErr error
 	for _, hf := range frame.Fields {
-		if hf.Name != "content-type" && hf.Name != ":status" && grpcErr != nil {
+		if hf.Name != "content-type" && hf.Name != ":status" && d.data.grpcErr != nil {
```

Comment:
Do we still need this?  Is it not safe to parse everything?  (If things must be skipped, can we move this logic to `processHeaderField`?)

If we're just skipping parsing some header fields in the name of efficiency, I don't think extra code complexity is worth optimizing error cases (unless it's a substantial difference).

Reply:
done